external/openglcts/modules/gl/gl3cTextureSizePromotion.cpp

```diff
@@ -244,6 +244,18 @@ void FunctionalTest::prepareSourceTexture(TextureInternalFormatDescriptor descri
 
 	if (!isTargetMultisampled(target))
 	{
+		glu::ContextType context_type = m_context.getRenderContext().getType();
+		if (isDepthType(descriptor) && glu::contextSupports(context_type, glu::ApiType::core(3, 1)))
+		{
+			/* 3.1 context may have GL_ARB_compatibility which has
+			 * GL_DEPTH_TEXTURE_MODE set to GL_LUMINANCE by default.
+			 * Set it to GL_RED since we expect depth texture sampling
+			 * to return vec4(depth, 0, 0, 1).
+			 */
+			gl.texParameteri(target, 0x884B, GL_RED);
```

Comment:
This magic number is a `GL_DEPTH_TEXTURE_MODE`, unfortunately VK-GL-CTS doesn't have header with it. What should be done in such case?

Reply:
When OpenGL 3.1 context is requested it may be created with `GL_ARB_compatibility` and from what I know there is no way to prevent this. In compatibility context the following is true:
```
The values of DEPTH_TEXTURE_MODE, TEXTURE_COMPARE_MODE,
  and TEXTURE_COMPARE_FUNC are LUMINANCE, NONE, and LEQUAL
  respectively.
```
So it has `GL_DEPTH_TEXTURE_MODE` set to `LUMINANCE` which breaks the test.

Reply:
@alegal-arm Did you have a chance to consider the above?

Reply:
We will look at this during the CTS group meeting. I can not say what the conclusion would be.